### PR TITLE
ADF-151. Fix compatibility with advanced search

### PR DESCRIPTION
**Pull request by voknil**
https://oat-sa.atlassian.net/browse/ADF-151

### CODE REVIEW

File: model/RevisionStorageInterface.php [hunk 1]
```diff
@@ -71,6 +71,14 @@ public function getData(Revision $revision);
      */
     public function getResourcesUriByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL);
 
+    /**
+     * @param string $query
+     * @param array $options
+     * @param string $predicate
```
Comment (boajer) [inline model/RevisionStorageInterface.php:77]:
```suggestion
```

File: model/RevisionStorageInterface.php [hunk 2]
```diff
@@ -71,6 +71,14 @@ public function getData(Revision $revision);
...
+     *
+     * @return array
+     */
+    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL);
```
Comment (boajer) [inline model/RevisionStorageInterface.php:81]:
```suggestion
    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array;
```

File: model/storage/RdsStorage.php [hunk 3]
```diff
@@ -271,6 +271,46 @@ public function getResourcesUriByQuery(string $query, array $options = [], strin
...
+        $order = isset($options['order']) ? strtoupper($options['order']) : ' ASC';
+
+        $queryBuilder->addOrderBy($sort, $order);
+        $queryBuilder->groupBy('rd.'.self::DATA_RESOURCE);
+
+        $result = $this->getPersistence()->query($queryBuilder->getSQL());
+        $resourcesUri = [];
+        /** @var Revision $statement */
+        while ($statement = $result->fetch()) {
+            $resourcesUri[] = [
```
Comment (boajer) [inline model/storage/RdsStorage.php:306]: Let's improve naming, but you may find something better then in suggestion
```suggestion
            $resourceData[] = [
```

File: model/RevisionStorageInterface.php [hunk 4]
```diff
@@ -71,6 +71,10 @@ public function getData(Revision $revision);
      */
     public function getResourcesUriByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL);
 
+    /**
+     * @return array
+     */
+    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array;
```
Comment (boajer) [inline model/RevisionStorageInterface.php:77]:
```suggestion
    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array;
```

as per squad conventions, we may omit phpdoc comments, which are not providing value on top of defined strong typing

File: model/storage/RdsStorage.php [hunk 5]
```diff
@@ -236,9 +237,55 @@ protected function saveData(Revision $revision, array $data)
...
+        );
+
+        $resourcesUri = [];
+        while ($statement = $result->fetch()) {
+            $resourcesUri[] = $statement[self::DATA_RESOURCE];
+        }
+        return $resourcesUri;
+    }
+
+    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array
```
Comment (boajer) [inline model/storage/RdsStorage.php:255]: Any reason to be public for this method? 

```suggestion
    private function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array
```

File: model/storage/RdsStorage.php [hunk 6]
```diff
@@ -236,9 +237,55 @@ protected function saveData(Revision $revision, array $data)
...
+                'label' => $statement[self::DATA_OBJECT],
+            ];
+        }
+        return $resourcesData;
+    }
+
+    /**
+     * @param string[] $selectedFields
+     */
+    public function getSelectedResourcesDataByQuery(
```
Comment (boajer) [inline model/storage/RdsStorage.php:278]: Any reason to be public? 
```suggestion
    private function getSelectedResourcesDataByQuery(
```

File: model/RevisionStorageInterface.php [hunk 7]
```diff
@@ -71,6 +71,7 @@ public function getData(Revision $revision);
      */
     public function getResourcesUriByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL);
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/RevisionStorageInterface.php:72]: this should be deprecated if no longer is being used or required

File: model/storage/RdsStorage.php (hunk 5 above)
Comment (mike-ionut-mihai-sandu-tao) [inline model/storage/RdsStorage.php:255]: please add tests

**Review by mike-ionut-mihai-sandu-tao — COMMENTED** [review]
lgtm, left some comments

**Review by boajer — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**